src/sentry/api/serializers/models/organization.py

```diff
@@ -653,10 +657,30 @@ def serialize(  # type: ignore[explicit-override, override]
             team__organization=obj
         ).count()
 
-        sample_rate = quotas.backend.get_blended_sample_rate(organization_id=obj.id)
-        context["isDynamicallySampled"] = (
-            has_dynamic_sampling(obj) and sample_rate is not None and sample_rate < 1.0
-        )
+        is_dynamically_sampled = False
+        sample_rate = None
+        if has_dynamic_sampling(obj):
+            if (
+                obj.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT)
+                == DynamicSamplingMode.ORGANIZATION
+            ):
+                sample_rate = obj.get_option(
+                    "sentry:target_sample_rate",
+                    quotas.backend.get_blended_sample_rate(organization_id=obj.id),
+                )
+                is_dynamically_sampled = sample_rate is not None and sample_rate < 1.0
+            else:
```

Comment:
This `else` assumes that we are only having Project sampling mode besides Organization, and that's fine for now, but this might change in the future if we add another dynamic sampling mode and this will be a potential bug